### ISSUE TITLE
ReplaceStringBuilderWithString should keep newlines

### ISSUE THREAD

**Issue by koppor** [openrewrite/rewrite-static-analysis#88]
```diff
-        String content = new StringBuilder()
-                .append(Localization.lang("A backup file for '%0' was found at [%1]",
+        String content = Localization.lang("A backup file for '%0' was found at [%1]",
                         originalPath.getFileName().toString(),
-                        backupFilename))
-                .append("\n")
-                .append(Localization.lang("This could indicate that JabRef did not shut down cleanly last time the file was used."))
-                .append("\n\n")
-                .append(Localization.lang("Do you want to recover the library from the backup file?"))
-                .toString();
+                        backupFilename) + "\n" + Localization.lang("This could indicate that JabRef did not shut down cleanly last time the file was used.") + "\n\n" + Localization.lang("Do you want to recover the library from the backup file?");
 ```

**Comment by timtebeek** [comment]
Sounds like we ought to pretend the white space prefix of the removed method calls to the method arguments. We could start with a unit test that replicates the issue, based on the above, and then step through with the debugger to figure out how best to add that in.

**Comment by pramud** [comment]
Hi @koppor I want to pick this up, but I am facing issues with setting up in my local and cannot find any documentation related to it, can you please or point me to any documentation if present.
Thanks,
Pramud

**Comment by timtebeek** [comment]
Hi @pramud ! Thank you for your offer to help; here's a few links which I hope are helpful in setting up your development environment:

1. https://docs.openrewrite.org/authoring-recipes/recipe-development-environment
2. https://github.com/openrewrite/rewrite/blob/main/CONTRIBUTING.md
3. https://docs.openrewrite.org/reference/building-openrewrite-from-source
4. https://github.com/openrewrite/rewrite-docs/issues/138

If you need any further help feel free to [join us on Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA).

**Comment by pramud** [comment]
I am debugging this, looks like this is not because of new lines but another method invocation inside the append.

**Comment by timtebeek** [comment]
Let us know if you'd need help with that @pramud ; I've not looked into the details yet. Welcome to open a draft pull request with only an unit test that replicates the issue as a start; that helps us all look at the same test failure.

**Comment by pramud** [comment]
As there were build issues when I tried to build locally, raised a PR in here 
https://github.com/openrewrite/rewrite/pull/3202

**Comment by timtebeek** [comment]
Yes we're in the middle of an upgrade to Rewrite 8.0; if you [check out this branch locally, and run `./gradlew pTML`](https://github.com/openrewrite/rewrite/tree/large-repositories) then the rewrite-static-analysis should pick that up out of your local `.m2/repository`. It's only this week that things are slightly different; once the 8.0 branch is merged, and especially once that's released things should become easier again.

**Comment by Jammy-Louie** [comment]
created a sample test to replicate

```
@Test
    void sample test() {
        rewriteRun(
          //language=java
          java(
            """
              class A {
                  void foo() {
                      String scenarioOne = new StringBuilder()
                          .append("A")
                          .append("B")
                          .append("C")
                          .toString();
                  }
              }
              """,
            """
              class A {
                  void foo() {
                      String scenarioOne = "A" +
                          "B" +
                          "C";
                  }
              }
              """
          )
        );
    }
```

it appears we could maintain the white space of the argument on line 147

https://github.com/openrewrite/rewrite-static-analysis/blob/6f2d4e1aaddd55e64713bc3e5353d7bfadfc27b3/src/main/java/org/openrewrite/staticanalysis/ReplaceStringBuilderWithString.java#L128-L149

The problem here is that when fetching the prefix on the `selectMethod`, white space is empty (`selectMethod.getPrefix().getWhitespace()`). However the spacing is available when inspecting the `JRightPadded` for the `selectMethod`

**Comment by timtebeek** [comment]
Hmm yes it seems you can access that white space by going through the Padding. That's an object that gives you access to the underlying non-code elements. Try using `method.getPadding().getSelect().getAfter()` to pass through the whitespace onto the arguments indeed.

It also seems as though we might be able to remove that `List<Expression> methodChain` completely; could be good to pick up too.
